### DISCUSSION TITLE
Great project; "support" for vSphere WebClient 6.0

### DISCUSSION THREAD

**Discussion by ytdlder** [ruffle-rs/ruffle#2491] [Ideas] [answered]
I'd hoped for some plugin like this; unfortunately "Action Script 3" isn't yet supported, which vSphere uses it seems.

This is the only thing why I still need Flash and I'm pretty sure I'll not be able to update the ESX to a newer version very soon...

So, might there be a chance that I could use your plugin to still be able to manage my VMs in the near future? ;-)

Thanks for your consideration!

**Comment by ROBERT-MCDOWELL** [comment, marked as answer]
you can help to contribute and accelerate the roadmap ;)

**Reply by ytdlder** [reply]
Would love to; but me not beeing a software developer... might not be the best idea ;-)

**Reply by ousia** [reply]
@ytdlder, just in case it might be relevant to you.

If your issues are critical, you might consider [sponsoring @ruffle-rs](https://github.com/sponsors/ruffle-rs).

That being written, I’m only an average computer user, no part of the developer team.

**Comment by InAUGral** [comment]
I found this project while trying to find an alternative for exactly this sort of webpage. If it could be added eventually that would certainly be handy. We have a few other monitoring tools at work in the same boat as vSphere client which require ActionScript3.

**Reply by ytdlder** [reply]
Well, I did find a solution: use an older Flash version, which doesn't "end"; we're using 32.0.0.371
At least in Opera I can still say to "use Flash" and after logging in (when the Flash parts are loaded) I can click "use old Flash this one time" or the like.
cheers

**Comment by n0samu** [comment]
These kinds of complex applications are the very most difficult to support. Plus Ruffle's focus is more on games than complex applications. I think Ruffle is still not close to being able support something like this.

**Reply by ROBERT-MCDOWELL** [reply]
but it would be logical to respect the API at the end right?

**Reply by n0samu** [reply]
Yes of course, the goal is to support everything, but some things will take a very long time

**Reply by ROBERT-MCDOWELL** [reply]
@n0samu 
is it because of technical aspects or missing rust libraries?

**Reply by n0samu** [reply]
I don't have any specific information about this since I don't have access to any vSphere software. I'm judging based on what I have seen in other complex Flex apps. They rely on many AVM2 functions that we have not begun work on yet. It's just the sheer API surface involved.
